src/jllvm/class/Descriptors.hpp

```diff
@@ -99,7 +99,7 @@ struct ArrayType
 
     bool operator==(const ArrayType& rhs) const
     {
-        return *componentType == *rhs.componentType;
```

Comment:
This change does not preserve semantics. The old one compares that these are structurally equivalent while with the changes it checks whether the pointers are identical